src/core/session.py

```diff
@@ -47,7 +49,17 @@ def onCreateSession(self, directory):
         self._followerTable = self.loadTable(SammoDataBase.FOLLOWER_TABLE_NAME)
         self._gpsTable = self.loadTable(SammoDataBase.GPS_TABLE_NAME)
 
-        QgsProject.instance().addMapLayer(self._gpsTable)
+        uri = (
+            "geopackage:"
+            + SammoDataBase.pathToDataBase(directory)
+            + "?projectName=sammo_boat_project"
+        )
+        if isNewDataBase:
+            # Save the QGIS projet into the database
+            QgsProject.instance().addMapLayer(self._gpsTable)
+            QgsProject.instance().write(uri)
```

Comment:
We should create a new project from scratch in this case instead of using the current one.

Reply:
Searching for the solution, I found the clear() method (from QgsProject class) which "_Clear the project - removes all settings and resets it back to an empty, default state._". I think it is the quickest way to achieve what you want.

Reply:
You can just create a whole new project with `project = QgsProject()` instead of using the current instance